pkg/controllers/dashboard/helm/repo_test.go

```diff
@@ -200,11 +212,39 @@ func TestShouldRefresh(t *testing.T) {
 			},
 			false,
 		},
+		{
+			"configmap - status contains a no-longer-existing configmap",
+			&catalog.RepoSpec{},
+			&catalog.RepoStatus{
+				IndexConfigMapName:      "missing-configmap",
+				IndexConfigMapNamespace: "cattle-system",
+				DownloadTime: metav1.Time{
+					Time: time.Now(),
+				},
+			},
+			true,
+		},
+		{
+			"configmap - status contains an existing configmap",
+			&catalog.RepoSpec{},
+			&catalog.RepoStatus{
+				IndexConfigMapName:      "configmap",
+				IndexConfigMapNamespace: "cattle-system",
+				DownloadTime: metav1.Time{
+					Time: time.Now(),
+				},
+			},
+			false,
+		},
 	}
 
+	mock := configMapMock{}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			result := shouldRefresh(tt.spec, tt.status)
+			rh := repoHandler{
+				configMaps: mock,
+			}
```

Comment:
Can this be moved outside the loop so that it is only initialized once?

Reply:
fixed.